hoomd/md/methods/rattle.py

```diff
@@ -0,0 +1,463 @@
+        the manifold. Even though the particles are mapped to the set surface
+        automatically, the mapping can lead to small inter-particle distances
+        and, hence, large forces between particles!
+
+    For the equations of motion, see:
+
+    * S. Paquay and R. Kusters  2016
+      (`paper link <https://doi.org/10.1016/j.bpj.2016.02.017>`__)
+
+    Note:
+        Users should use the subclasses and not instantiate `MethodRATTLE`
+        directly.
+    """
+
+    def __init__(self, manifold_constraint, tolerance):
+
+        param_dict = ParameterDict(manifold_constraint=OnlyTypes(
+            Manifold, allow_none=False),
+                                   tolerance=float(tolerance))
+        param_dict['manifold_constraint'] = manifold_constraint
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _add(self, sim):
+        self.manifold_constraint._add(sim)
+        super()._add(sim)
+
+    def _attach_constraint(self, sim):
+        if not self.manifold_constraint._attached:
+            self.manifold_constraint._attach()
+
+    def _getattr_param(self, attr):
+        if self._attached:
+            if attr == "manifold_constraint":
+                return self._param_dict["manifold_constraint"]
+            parameter = getattr(self._cpp_obj, attr)
+            return parameter
+        else:
+            return self._param_dict[attr]
+
+    def _setattr_param(self, attr, value):
+        if attr == "manifold_constraint":
+            raise AttributeError(
+                "Cannot set manifold_constraint after construction.")
+        super()._setattr_param(attr, value)
+
+
+class NVE(MethodRATTLE):
+    r"""NVE Integration via Velocity-Verlet with RATTLE constraint.
+
+    Args:
+        filter (`hoomd.filter.ParticleFilter`): Subset of particles on which to
+         apply this method.
+
+        manifold_constraint (:py:mod:`hoomd.md.manifold.Manifold`): Manifold
+            constraint.
+
+        limit (None or `float`): Enforce that no particle moves more than a
+            distance of a limit in a single time step. Defaults to None
+
+        tolerance (`float`): Defines the tolerated error particles are
+            allowed to deviate from the manifold in terms of the implicit
+            function.  This is only used if RATTLE algorithm is triggered.
+            Defaults to 1e-6
+
+    `NVE` performs constant volume, constant energy simulations using
+    the standard Velocity-Verlet method. The particles are constrained to a
+    manifold by using the RATTLE algorithm. For poor initial conditions that
+    include overlapping atoms, a limit can be specified to the movement a
+    particle is allowed to make in one time step. After a few thousand time
+    steps with the limit set, the system should be in a safe state to continue
+    with unconstrained integration.
+
+    .. todo::
+        Update when zero momentum updater is added.
```

Comment:
Simplify this discussion. There is no need to repeat text from `md.methods.NVE`. Instead, what NVE is briefly and that this variant constrains particles to the manifold using the RATTLE algorithm. Then link to `md.methods.NVE` in a seealso block for details.